Python Help Window: Update dependencies and conditionalize QtAssistant build in docs & conda recipe

darshdinger opened the pull request:
### Description of work

#### Summary of work
<!-- Please provide a short, high level description of the work that was done.
-->
This work updates the conda packaging configuration for `mantiddocs`. Specifically, it modifies the dependencies listed in `conda/recipes/mantiddocs/meta.yaml` to ensure the correct versions and packages (like `texlive-core`) are used during the build. It also adjusts the CMake logic within `docs/CMakeLists.txt` related to the optional building of QtAssistant/QtHelp documentation (`ENABLE_QTASSISTANT`), ensuring it integrates correctly with the packaging process and reflects the updated dependencies. Minor corresponding changes were made to `conda/recipes/mantiddocs/build.sh`.
<!-- Why has this work been done? If there is no linked issue please provide appropriate context for this work.
#### Purpose of work
This can be removed if a github issue is referenced below
-->
This work updates the conda packaging configuration for `mantiddocs`. Specifically, it modifies the dependencies listed in `conda/recipes/mantiddocs/meta.yaml` to ensure the correct versions and packages (like `texlive-core`) are used during the build. It also adjusts the CMake logic within `docs/CMakeLists.txt` related to the optional building of QtAssistant/QtHelp documentation (`ENABLE_QTASSISTANT`), ensuring it integrates correctly with the packaging process and reflects the updated dependencies. Minor corresponding changes were made to `conda/recipes/mantiddocs/build.sh`.
Issue #37248 . <!-- and fix #xxxx or close #xxxx xor resolves #xxxx. One line per issue fixed. -->
<!-- alternative
*There is no associated issue.*
-->

<!-- If the original issue was raised by a user they should be named here. Do not leak email addresses
**Report to:** [user name]
-->

#### Further detail of work
<!-- Please provide a more detailed description of the work that has been undertaken.
-->
1.  **`conda/recipes/mantiddocs/meta.yaml`:**
    *   Updated the `requirements: build:` section to include `texlive-core` as a dependency, necessary for rendering math equations via `sphinx.ext.imgmath`.
    *   Refreshed other dependencies like `sphinx`, `sphinx_bootstrap_theme`, `mantidqt`, `versioningit`, `graphviz`, and `doxygen` to ensure appropriate versions are pulled during the conda build. (Adjust this point if specific version changes were the focus).

2.  **`docs/CMakeLists.txt`:**
    *   Modified the logic controlling the generation of Qt Help files (`.qhc`, `.qch`). The condition for adding the `docs-qthelp` custom target and associated `qcollectiongenerator` command is now explicitly tied to the `ENABLE_QTASSISTANT` CMake option, in addition to the `DOCS_QTHELP` check. Previously, parts of this logic might have executed even if `ENABLE_QTASSISTANT` was off.
    *   Updated the `PACKAGE_DOCS` section to also check for `ENABLE_QTASSISTANT` before attempting to install the Qt Help files (`.qhc`, `.qch`, images, static files) into the package. This ensures these files are only included in the final package when Qt Assistant support is explicitly enabled during the build configuration.

3.  **`conda/recipes/mantiddocs/build.sh`:**
    *   Minor adjustments were made to the CMake arguments passed during the build to align with the changes in `docs/CMakeLists.txt` and potentially reflect the updated dependencies or build environment variables. (You might want to specify exactly which arguments were added/changed if known, e.g., explicitly setting `-DENABLE_QTASSISTANT=OFF` if that was done).


Here is the markdown embedded build status: [![Build Status](https://builds.mantidproject.org/job/build_packages_from_branch/1130/badge/icon)](https://builds.mantidproject.org/job/build_packages_from_branch/1130/)

### To test:
Tested by running the build script.
<!-- Instructions for testing.
There should be sufficient instructions for someone unfamiliar with the application to test - unless a specific
reviewer is requested.
If instructions for replicating the fault are contained in the linked issue then it is OK to refer back to these.
-->

<!-- delete this if you added release notes
*This does not require release notes* because **fill in an explanation of why**
If you add release notes please save them as a separate file using the Issue or PR number as the file name. Check the file is located in the correct directory for your note(s).
-->

<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

darshdinger committed:
Updates for packaging

darshdinger committed:
Updates for packaging

darshdinger committed:
Updates for packaging

darshdinger committed:
Updates for packaging

darshdinger committed:
Updates for packaging

darshdinger committed:
Updates for packaging

darshdinger committed:
Updates for packaging

darshdinger committed:
Updates for packaging

darshdinger committed:
Resolve Jenkins build error

darshdinger committed:
Resolve Jenkins build error

darshdinger committed:
Resolve Jenkins build error

darshdinger committed:
Resolve Jenkins build error

darshdinger committed:
Resolve Jenkins build error

darshdinger committed:
Resolve Jenkins build error

peterfpeterson commented on conda/recipes/mantiddocs/meta.yaml:
```diff
@@ -36,6 +36,9 @@ requirements:
     - sphinx_bootstrap_theme {{ sphinx_bootstrap_theme }}
     - mantidqt {{ version }}
     - versioningit {{ versioningit }}
+    - graphviz {{ graphviz }}
```
What of these are still needed?

thomashampson replied:
Currently we install LaTeX in our linux docker image because texlive_core was missing at least one package that we required (see [here](https://github.com/mantidproject/dockerfiles/blob/28beddf097bae984e23e83f8c3487bb370737056/Linux/development/docker/CentOS7Slim.Dockerfile#L41)).
If texlive-core now works for us that would be great.

peterfpeterson replied:
I don't think we should be running latex anymore at all. That was used for generating the equations in the static HTML that got compiled for qtassistant. The dynamic pages are using mathjax (javascript).

darshdinger committed:
Uncomment QT_QPA_PLATFORM

darshdinger committed:
Uncomment QT_QPA_PLATFORM

darshdinger committed:
Uncomment QT_QPA_PLATFORM

darshdinger committed:
Uncomment QT_QPA_PLATFORM

darshdinger committed:
Uncomment QT_QPA_PLATFORM

darshdinger committed:
Uncomment QT_QPA_PLATFORM

darshdinger committed:
Remove unnecessary host requirements

darshdinger committed:
Remove unnecessary host requirements

darshdinger committed:
Remove unnecessary host requirements

darshdinger committed:
Remove unnecessary host requirements

darshdinger committed:
Remove unnecessary host requirements

darshdinger committed:
Remove unnecessary host requirements

darshdinger committed:
Added texlive-core back to meta.yaml

darshdinger committed:
Added texlive-core back to meta.yaml

darshdinger committed:
Added texlive-core back to meta.yaml

darshdinger committed:
Added texlive-core back to meta.yaml

darshdinger committed:
Added texlive-core back to meta.yaml

darshdinger committed:
Added texlive-core back to meta.yaml

darshdinger committed:
Remove SPHINX_WARNINGS_AS_ERRORS

darshdinger committed:
Remove SPHINX_WARNINGS_AS_ERRORS

darshdinger committed:
Remove SPHINX_WARNINGS_AS_ERRORS

darshdinger committed:
Remove SPHINX_WARNINGS_AS_ERRORS

darshdinger committed:
Remove SPHINX_WARNINGS_AS_ERRORS

darshdinger committed:
Remove SPHINX_WARNINGS_AS_ERRORS

backmari commented:
I have tested the Python Help Window (offline version) and it works as expected for me. For the build, just running the script didn't work for me. Can you please give step-by-step instructions? If someone else has tested the build, or it is part of the CI, that's good enough for me.

When using `-DENABLE_QTASSISTANT=OFF` the docs open in the "Python Help Window" and the indicator says "offline".
When using `-DENABLE_QTASSISTANT=ON` the docs open in the browser.
Is this as expected? How do you use the "Python Help Window" in "online" mode?

darshdinger commented:
> I have tested the Python Help Window (offline version) and it works as expected for me. For the build, just running the script didn't work for me. Can you please give step-by-step instructions? If someone else has tested the build, or it is part of the CI, that's good enough for me.
> 
> When using `-DENABLE_QTASSISTANT=OFF` the docs open in the "Python Help Window" and the indicator says "offline". When using `-DENABLE_QTASSISTANT=ON` the docs open in the browser. Is this as expected? How do you use the "Python Help Window" in "online" mode?

Thanks for testing this Marie!! I appreciate it greatly. Glad to hear it works for you as it should.

So you are seeing the `offline` indicator which essentially means that you have the mantid docs built locally and they are found within your documentation build tree. In order to see the `online` docs, you will need to clear your doc build tree to insure that no documentation is present on your local machine. This new help window will revert automatically to using online documentation showing the `online` indicator in this case.

Building mantid with the `-DENABLE_QTASSISTANT=ON` flag enables you to use the old implementation of the python help interface which seemingly defaults to opening a web browser if local documentation is not built. I've found through thorough testing that this old interface it a little unreliable in locating local docs and usually defaults to opening a web browser. It was challenging for me to get the old `qtassistant` based interface to start.

This new implementation was purposefully done in a "side-by-side" way in which the old help window is enabled if you just build mantid the "normal" way (i.e. using the `ninja` command) . This was done as to not hinder mantid's functionality when developing this new `qtwebengine` based version of the help window thus requiring you to use the compiler flag `-DENABLE_QTASSISTANT=OFF` upon building mantid. Hope this makes sense.

This PR, once it is complete, will enable the use of the new version of the help window upon building mantid the "normal" way as stated before instead of having to use the compiler flag to enable it.

In order to run a local build of the conda package you need to start in your local mantid repo (for me this is just `/home/dzj/mantid/`) and insure that your conda env is just base as you do not need to have `mantid-developer` conda env activated. Then you can use the command: `./buildconfig/Jenkins/Conda/package-conda /tmp/mantid-build-test/ --build-mantid --build-qt --build-docs` to run the conda build script in order to kickoff the build of a new conda package locally. Hope this answers your questions. Let me know if there is anymore confusion.

peterfpeterson commented:
I tried the package from
```sh
conda  create -n mantidhelp mantid-ornl/label/nightly::mantidworkbench -c mantid
```
which was version 6.12.20250513.1601. There is an issue hiding in the build script because opening the help (from menu item) writes out the log messages
```
MantidHelpWindow-[Information] Failed to find help system collection file "MantidProject.qhc"
MantidHelpWindow-[Information] Without collection file redirecting help to default web browser
```
and uses the system browser to point at the online documentation.

This suggests the package build is still using the old qtassistant help window by default.

peterfpeterson commented on docs/CMakeLists.txt:
```diff
@@ -120,19 +120,21 @@ if(DOCS_QTHELP)
     REQUIRED
   )
   add_sphinx_build_target(qthelp ${DOCS_MATH_EXT} docs-qtassistant qthelp)
```
I think this needs to get moved inside of the `if (ENABLE_QTASSISTANT)` section

darshdinger committed:
Fixes

darshdinger committed:
Fixes

darshdinger committed:
Fixes

darshdinger committed:
Fixes

darshdinger committed:
Fixes

darshdinger committed:
Fixes

darshdinger committed:
Fixes

darshdinger committed:
Fixes

darshdinger committed:
Move add_sphinx_build_target

darshdinger committed:
Move add_sphinx_build_target

darshdinger committed:
Move add_sphinx_build_target

darshdinger committed:
Move add_sphinx_build_target

darshdinger committed:
Move add_sphinx_build_target

darshdinger committed:
Move add_sphinx_build_target

peterfpeterson reviewed (changes requested):
So close. It does build/start the helpwindow, but it installs the HTML docs in the directory `envs/<envname>/share/doc/` rather than `envs/<envname>/share/doc/html/` which is where the property is looking for it. 

You can verify this by downloading the mantiddocs package and running
```sh
tar tjf Downloads/mantiddocs-*.tar.bz2
```

sf1919 commented:
Has this been rebased since we moved to Python 3.11? I ask because the osx failure seems to be pulling in v3.10 of Python which it shouldn't.

darshdinger committed:
Fix location of docs install

darshdinger committed:
Fix location of docs install

darshdinger committed:
Fix location of docs install

darshdinger committed:
Fix location of docs install

darshdinger committed:
Fix location of docs install

darshdinger committed:
Fix location of docs install

darshdinger committed:
Fix location of docs install

darshdinger committed:
Fix location of docs install

darshdinger committed:
Revert previous and change meta.yaml

darshdinger committed:
Revert previous and change meta.yaml

darshdinger committed:
Revert previous and change meta.yaml

darshdinger committed:
Revert previous and change meta.yaml

darshdinger committed:
Revert previous and change meta.yaml

darshdinger committed:
Revert previous and change meta.yaml

darshdinger committed:
Revert previous and change meta.yaml

darshdinger committed:
Revert previous and change meta.yaml

darshdinger committed:
Reference mantid code base

darshdinger committed:
Reference mantid code base

darshdinger committed:
Reference mantid code base

darshdinger committed:
Reference mantid code base

darshdinger committed:
Reference mantid code base

darshdinger committed:
Reference mantid code base

darshdinger committed:
Reference mantid code base

darshdinger committed:
Reference mantid code base

darshdinger committed:
Update CMakeLists.txt

darshdinger committed:
Update CMakeLists.txt

darshdinger committed:
Update CMakeLists.txt

darshdinger committed:
Update CMakeLists.txt

darshdinger committed:
Update CMakeLists.txt

darshdinger committed:
Update CMakeLists.txt

darshdinger committed:
Change docs install location from share/doc to share/doc/html

darshdinger committed:
Change docs install location from share/doc to share/doc/html

darshdinger committed:
Change docs install location from share/doc to share/doc/html

darshdinger committed:
Change docs install location from share/doc to share/doc/html

darshdinger committed:
Change docs install location from share/doc to share/doc/html

darshdinger committed:
Change docs install location from share/doc to share/doc/html

darshdinger committed:
Change docs install location from share/doc to share/doc/html

darshdinger committed:
Change docs install location from share/doc to share/doc/html

darshdinger committed:
Fix HelpWindow warning

darshdinger committed:
Fix HelpWindow warning

darshdinger committed:
Fix HelpWindow warning

darshdinger committed:
Fix HelpWindow warning

darshdinger committed:
Fix HelpWindow warning

darshdinger committed:
Fix HelpWindow warning

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

peterfpeterson commented on qt/python/mantidqt/mantidqt/gui_helper.py:
```diff
@@ -80,7 +80,7 @@ def __get_collection_file(collection_file: str) -> str:
             return "HELP COLLECTION FILE NOT FOUND"
         else:
-            collection_file = os.path.join(__mtd_bin_dir, "../docs/qthelp/MantidProject.qhc")
+            collection_file = os.path.join(__mtd_bin_dir, "..", "share", "doc", "html", "MantidProject.qhc")
```
It doesn't need to be changed since the qhc file is going away altogether soon, but why did the path change?

darshdinger commented on conda/recipes/mantiddocs/build.sh:
```diff
@@ -16,6 +16,9 @@ cmake \
   -DMANTID_QT_LIB=SYSTEM \
   -DENABLE_WORKBENCH=OFF \
   -DENABLE_DOCS=ON \
+  -DSPHINX_WARNINGS_AS_ERRORS=OFF \
```
I was using this to allow me to build without more headaches. Will remove this from all build.sh scripts.

darshdinger commented on conda/recipes/mantidworkbench/build.sh:
```diff
@@ -16,7 +16,10 @@ cmake \
   -DMANTID_QT_LIB=SYSTEM \
   -DENABLE_WORKBENCH=ON \
+  -DENABLE_QTASSISTANT=OFF \
+  -DSPHINX_WARNINGS_AS_ERRORS=OFF \
```
Removed. :)

darshdinger committed:
Updates based on comments

darshdinger committed:
Updates based on comments

darshdinger committed:
Updates based on comments

darshdinger committed:
Updates based on comments

darshdinger committed:
Updates based on comments

darshdinger committed:
Updates based on comments

darshdinger committed:
Updates based on comments

darshdinger committed:
Updates based on comments

darshdinger committed:
Merge branch 'python-help-window-update-packaging' of https://github.com/mantidproject/mantid into python-help-window-update-packaging

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

rosswhitfield committed:
Cleanup NeXus::File forward declarations (#39420)

### Description of work

Following #39414 this cleans up some missed forward declarations for
`NeXus::File`.

#### Summary of work
<!-- Please provide a short, high level description of the work that was
done.
-->

<!-- Why has this work been done? If there is no linked issue please
provide appropriate context for this work.
#### Purpose of work
This can be removed if a github issue is referenced below
-->

Ref https://github.com/mantidproject/mantid/issues/38332

<!-- If the original issue was raised by a user they should be named
here. Do not leak email addresses
**Report to:** [user name]
-->

#### Further detail of work
<!-- Please provide a more detailed description of the work that has
been undertaken.
-->

### To test:

Simple change, code review should suffice.

Commit:
Change Nexus path to address (#39417)

### Description of work

To help maintain good order and civility, `path` has been replaced with
`address` whenever referring to a location within a Nexus file.

<!-- Please provide a short, high level description of the work that was
done.
-->

<!-- Why has this work been done? If there is no linked issue please
provide appropriate context for this work.
#### Purpose of work
This can be removed if a github issue is referenced below
-->

Done in support of #38332 <!-- and fix #xxxx or close #xxxx xor resolves
#xxxx. One line per issue fixed. -->
<!-- alternative
*There is no associated issue.*
-->

<!-- If the original issue was raised by a user they should be named
here. Do not leak email addresses
**Report to:** [user name]
-->

<!-- Please provide a more detailed description of the work that has
been undertaken.
-->

### To test:

This is a refactor to rename variables and functions. All existing tests
should pass.

*This does not require release notes* because this is an internal change
only.

peterfpeterson committed:
Another batch of nexus changes into ornl-next (#39422)

This brings the following changes into `ornl-next`
* #39412 
* #39420 
* #39417

---------

Co-authored-by: Ross Whitfield <whitfieldre1@ornl.gov>
Co-authored-by: Carson Sears <searscr@ornl.gov>

mantid-builder committed:
Merge ornl-next into main

peterfpeterson commented:
Local build works and installing the package
```sh
conda create -n mantidhelp -c mantid mantid-ornl/label/unstable::mantidworkbench
conda activate mantidhelp
mantidworkbench
```
works. The only question is where are the release notes.

darshdinger committed:
Add release notes.

darshdinger committed:
Add release notes.

darshdinger committed:
Add release notes.

darshdinger committed:
Add release notes.

darshdinger committed:
Add release notes.

darshdinger committed:
Add release notes.

darshdinger committed:
Add release notes.

darshdinger committed:
Add release notes.

darshdinger committed:
Merge branch 'python-help-window-update-packaging' of https://github.com/mantidproject/mantid into python-help-window-update-packaging

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

darshdinger committed:
Added release notes for 6.14 release

darshdinger committed:
Added release notes for 6.14 release

darshdinger committed:
Added release notes for 6.14 release

darshdinger committed:
Added release notes for 6.14 release

darshdinger committed:
Added release notes for 6.14 release

darshdinger committed:
Added release notes for 6.14 release

darshdinger committed:
Added release notes for 6.14 release

darshdinger committed:
Added release notes for 6.14 release

darshdinger committed:
Merge branch 'python-help-window-update-packaging' of https://github.com/mantidproject/mantid into python-help-window-update-packaging

darshdinger committed:
Move release notes

darshdinger committed:
Move release notes

darshdinger committed:
Move release notes

darshdinger committed:
Move release notes

darshdinger committed:
Move release notes

darshdinger committed:
Move release notes

darshdinger committed:
Move release notes

darshdinger committed:
Move release notes

MohamedAlmaki committed:
Handle import errors when simpleapi modules are imported without matplotlib installed (#39421)

By default, matplotlib is not installed when you install the Mantid Python library.
This will lead to import errors when attempting to access a module in the Framework,
as some of the algorithms use matplotlib. The solution here is to add dummy
functions that will raise an exception when called. This will delay the
exception handling from compile time to run time. Additionally, I added
a user prompt to notify the user to install matplotlib if they wish to
use these algorithms.

Fixes #39396. <!-- and fix #xxxx or close #xxxx xor resolves #xxxx. One
line per issue fixed. -->

### To test:
1- Build Mantid
2- In the conda environment, uninstall matplotlib "pip uninstall
matplotlib"
3- Run the following Python commands in the Python shell:
```
from mantid.simpleapi import plotSpectrum
plotSpectrum()
```
4- This will raise an exception and notify the user to install
matplotlib
5- Make sure other modules that doesn't depend on matplotlib works fine,
for example run:
```
from mantid.api import FrameworkManager
FrameworkManager.Instance()
```
6- It should work fine without errors as the FrameworkManager doesn't
require matplotlib


<!-- Instructions for testing.
There should be sufficient instructions for someone unfamiliar with the
application to test - unless a specific
reviewer is requested.
If instructions for replicating the fault are contained in the linked
issue then it is OK to refer back to these.
-->

<!-- delete this if you added release notes
*This does not require release notes* because **fill in an explanation
of why**
If you add release notes please save them as a separate file using the
Issue or PR number as the file name. Check the file is located in the
correct directory for your note(s).
-->

<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!
-->

---

### Reviewer

Please comment on the points listed below ([full
description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so
please comment clearly on what you have checked during your review.** If
changes are made to the PR during the review process then your final
comment will be the most important for gatekeepers. In this comment you
should make it clear why any earlier review is still valid, or confirm
that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding
standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI
standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the
changes appropriately?
- Do the release notes conform to the [release notes
guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the
tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been
added/updated?

Does everything look good? Mark the review as **Approve**. A member of
`@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set
the review status to "Request changes". This will stop the PR from
showing up in the list for other gatekeepers.

mantid-builder committed:
Merge release-next into main

peterfpeterson committed:
Remove NXinquirefile from nexus api (#39424)

Since nexus is relying on hdf5 internals for the functionality of
linking to data in external files, this does not need to exist in the
nexus api layer.

This also removes a function declaration where the implementation was
previously removed.

Refs #38332 and
[EWM11399](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=11399)

### To test:

Removing unused functions are fun! The builds pass means that the
methods were correctly removed.

*This does not require release notes* because it removes unused internal
functions.

<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!
-->

---

### Reviewer

Please comment on the points listed below ([full
description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so
please comment clearly on what you have checked during your review.** If
changes are made to the PR during the review process then your final
comment will be the most important for gatekeepers. In this comment you
should make it clear why any earlier review is still valid, or confirm
that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding
standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI
standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the
changes appropriately?
- Do the release notes conform to the [release notes
guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the
tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been
added/updated?

Does everything look good? Mark the review as **Approve**. A member of
`@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set
the review status to "Request changes". This will stop the PR from
showing up in the list for other gatekeepers.

dlcaballero16 committed:
Fix for hidden workspaces not being hidden in a group (#39402)

This fixes an issue where when a hidden workspace (name starts with
"__") is in a WorkspaceGroup, the workspace would still show up in the
workspace browser.

Co-authored-by: Pete Peterson <petersonpf@ornl.gov>

darshdinger committed:
Merge branch 'python-help-window-update-packaging' of https://github.com/mantidproject/mantid into python-help-window-update-packaging

darshdinger committed:
Merge branch 'python-help-window-update-packaging' of https://github.com/mantidproject/mantid into python-help-window-update-packaging
